xenmgr/XenMgr/PowerManagement.hs

```diff
@@ -320,12 +339,18 @@ executePmActionInternal action supervised =
     when (action /= ActionNothing) $ do
       info $ "PM: received pm action request: " ++ show action
       current <- liftRpc getCurrentPmAction
-      case current of
-        Just c  -> info $ "PM: but pm action " ++ show c ++ " is currently running, so cannot do"
-        Nothing -> (do liftRpc $ setCurrentPmAction action
+      case (action, current) of
+        (ActionIdleShutdown, Just ActionIdleShutdown) -> info $ "PM: but pm action idle-shutdown is already running, so not doing"
```

Comment:
"PM: but pm action idle-shutdown is already running - do nothing"

Reply:
Yea I like that, will update.